README.md

```diff
@@ -39,9 +39,11 @@ Read ["Installing Rust"] from [The Book].
     $ make && make install
     ```
 
-    > ***Note:*** You may need to use `sudo make install` if you do not
-    > normally have permission to modify the destination directory. The
-    > install locations can be adjusted by passing a `--prefix` argument
+    > ***Note:*** Currently `sudo make install` is not supported, so if you
+    > do not have permission to modify the destination directory then you'll
+    > want to install to a local directory like `$HOME/rust`.
```

Comment:
Ah so I'm less concerned about the build directory today (which we will hose over with `root` owned files if you `sudo make install` in a fresh dir). What I'm worried about here is that Cargo access global state, like your home directory, where a `sudo make install` could damage quite a bit.

So basically tl;dr; is that `sudo cargo build` is likely to go terribly wrong (although I haven't tried to hard to verify this). Transitively this would imply that `sudo make install` would also go terribly wrong as it involves a `cargo build` along the way.

I wonder if though we could just detect this in rustbuild and avoid running Cargo at all...